src/components/avatar/__tests__/Avatar-test.js

```diff
@@ -66,18 +72,27 @@ describe('components/avatar/Avatar', () => {
             avatarUrl: 'http://foo.bar/baz123_invalid',
         };
 
-        const wrapper = shallow(<Avatar {...props} />);
-
-        wrapper.instance().onImageError();
-        expect(wrapper.state('hasImageErrored')).toEqual(true);
+        let wrapper;
+        act(() => {
+            wrapper = mount(<Avatar {...props} />);
+        });
+        expect(wrapper.find(MockAvatarImage).length).toEqual(1);
 
-        wrapper.setProps({
-            ...props,
-            avatarUrl: 'http://foo.bar/baz123_invalid_new',
+        act(() => {
+            const avatarImage = wrapper.find(MockAvatarImage);
+            avatarImage.prop('onError')();
```

Comment:
It looks like this is manually invoking the `onError` prop callback? Can we use `.simulate` or `.simulateError`?

Reply:
`enzyme` devs advocate against using `simulate`. They actually recommend calling the prop directly.

Reply:
https://github.com/airbnb/enzyme/issues/1606